app/models/profile.rb

```diff
@@ -47,19 +47,22 @@ def compliant?(host)
   def results(host)
     Rails.cache.fetch("#{id}/#{host.id}/results", expires_in: 1.week) do
       rule_results = RuleResult.find_by_sql(
-        ['SELECT rule_results.* FROM (
-          SELECT rr2.*,
-             rank() OVER (
-                    PARTITION BY rule_id, host_id
-                    ORDER BY end_time DESC, created_at DESC
-             )
-          FROM rule_results rr2
-          WHERE rr2.host_id = ? AND rr2.rule_id IN
-             (SELECT rules.id FROM rules
-              INNER JOIN profile_rules
-              ON rules.id = profile_rules.rule_id
-              WHERE profile_rules.profile_id = ?)
-       ) rule_results WHERE RANK = 1', host.id, id]
+        [
+          'SELECT rule_results.* FROM (
+           SELECT rr2.*,
+              rank() OVER (
+                     PARTITION BY rule_id, host_id
+                     ORDER BY end_time DESC, created_at DESC
+              )
+           FROM rule_results rr2
+           WHERE rr2.host_id = ? AND rr2.result IN (?) AND rr2.rule_id IN
+              (SELECT rules.id FROM rules
+               INNER JOIN profile_rules
+               ON rules.id = profile_rules.rule_id
+               WHERE profile_rules.profile_id = ?)
+          ) rule_results WHERE RANK = 1',
+          host.id, RuleResult::SELECTED, id
+        ]
```

Comment:
This could go into a scope as well, it would allow for a descriptive name and would highlight the parameters passed.

Reply:
IMO scopes should return an AR::Relation so they can be chained